Install pre-compiled OpenCV4 for Swift notebooks with OpenCV

vvmnnnkv opened the pull request:
This change adds pre-built OpenCV v4 in Colab environment to solve problem with SwiftCV installation in some Swift notebooks:  https://forums.fast.ai/t/opencv-in-swift/44514/26

Details:
1. Added script `compile_colab.sh` to build and package opencv4 (this is for reference)
2. Using this script, `https://storage.googleapis.com/opencv4/opencv4.tgz` is created (using CPU Colab runtime)
3. `colab` script is updated to download & use compiled opencv4 binary

I've tested that SwiftCV notebook works when this binary is installed.

Note: it would be good to upload binary to other location (e.g. `https://course.fast.ai/setup/`) instead of `https://storage.googleapis.com/opencv4/opencv4.tgz` because this is my GCP bucket and I may eventually remove it.

jph00 commented:
I think an extra 7s on every colab start isn't worth it for this, since most people using fastai on colab won't be using it. How about instead putting the opencv setup steps into a separate bash script, and then in a swiftcv notebook add a cell saying "colab users - install open cv by commenting this" and have the necessary code to download/run the script there?

vvmnnnkv commented:
Make sense, moved code to separate script `nbs/swift/SwiftCV/install/install_colab.sh` and added `%system SwiftCV/install/install_colab.sh` in the first cell before `%install ...` in all notebooks that use opencv (assuming that course-v3/nbs/swift/ will be installed into /content/). Install script checks if opencv is already installed in VM, so it should install only once per new VM.

jph00 commented:
I think that cell needs to be either commented out, with a comment above to say "uncomment this to use on colab", or in some other ways needs to be set up to only run on colab.

vvmnnnkv commented:
Oops, totally forgot about non-Colab usage! :)
Commented install command like you suggested.

jph00 commented:
Thanks!
